### PR TITLE
Move test resources to test directory.

### PR DESCRIPTION
**Pull request by ming13**
Addresses #463.

### CODE REVIEW

**Comment by barbeau** [comment]
@ming13 Thanks for the PR! 

IIRC these files are used in unit tests so simply deleting them will break the tests. I believe the best fix for this is to move them into the test directory as mentioned in https://github.com/googlemaps/android-maps-utils/issues/463#issuecomment-410360845.

**Comment by ming13** [comment]
@barbeau, yeah, I read about that but neither code search or Save Delete option in IJ have found any usage. Renamed the PR and adjusted the change.

BTW, does anybody run tests? I had to fix a bunch of them to be green. And CI is not running them (and running at all as it seems). This is scary.

File: library/build.gradle [hunk 1]
```diff
@@ -34,6 +35,9 @@ android {
         installOptions "-d", "-t"
     }
 
+    defaultConfig {
+        testInstrumentationRunner 'android.support.test.runner.AndroidJUnitRunner'
+    }
```
Comment (ming13) [inline library/build.gradle]: Tests were not found without the runner. No idea how it worked before (if it did).

File: library/tests/src/com/google/maps/android/clustering/StaticClusterTest.java [hunk 2]
```diff
@@ -19,13 +19,15 @@
...
 
+import org.junit.Before;
 import org.junit.Test;
 import org.junit.Assert;
 
 public class StaticClusterTest {
 
     private StaticCluster<ClusterItem> mCluster;
 
+    @Before
```
Comment (ming13) [inline library/tests/src/com/google/maps/android/clustering/StaticClusterTest.java]: `setUp` was not called without the annotation.

File: library/tests/src/com/google/maps/android/data/LineStringTest.java [hunk 3]
```diff
@@ -48,19 +48,19 @@ public void testGetGeometryObject() throws Exception {
...
-        Assert.assertEquals(lineString.getGeometryObject().get(0).latitude, 90.0);
-        Assert.assertEquals(lineString.getGeometryObject().get(1).latitude, 90.0);
-        Assert.assertEquals(lineString.getGeometryObject().get(2).latitude, 90.0);
-        Assert.assertEquals(lineString.getGeometryObject().get(3).longitude, 53.0);
-        Assert.assertEquals(lineString.getGeometryObject().get(4).longitude, 54.0);
+        Assert.assertEquals(lineString.getGeometryObject().get(0).latitude, 90.0, 0);
+        Assert.assertEquals(lineString.getGeometryObject().get(1).latitude, 90.0, 0);
+        Assert.assertEquals(lineString.getGeometryObject().get(2).latitude, 90.0, 0);
+        Assert.assertEquals(lineString.getGeometryObject().get(3).longitude, 53.0, 0);
+        Assert.assertEquals(lineString.getGeometryObject().get(4).longitude, 54.0, 0);
```
Comment (ming13) [inline library/tests/src/com/google/maps/android/data/LineStringTest.java]: This and a bunch of similar changes address this:

```
java.lang.AssertionError: Use assertEquals(expected, actual, delta) to compare floating-point numbers
    at org.junit.Assert.fail(Assert.java:88)
    at org.junit.Assert.assertEquals(Assert.java:667)
    at org.junit.Assert.assertEquals(Assert.java:656)
```

**Comment by barbeau** [comment]
CI issue is discussed in https://github.com/googlemaps/android-maps-utils/issues/371.

Yes, the project probably needs some updating to run tests correctly in Android Studio.  Let's split those changes out to their own PRs though so each serves it's own purpose.

Looking at my old comment the KML files are used in `library/tests/src/com/google/maps/android/kml/Kml*Test.java` classes - I don't believe Android Studio will flag them via a Find on the files.

**Comment by ming13** [comment]
@barbeau, maybe it’ll be fine to merge moving files and fixing tests together? I mean, it literally makes tests runnable, it wouldn’t be even possible to merge this PR if the CI would be configured properly to run tests.

**Comment by ming13** [comment]
@jpoehnelt, @friederbluemle, please take a look. We had open questions so I’m just waiting for the clarification.

**Comment by jpoehnelt** [comment]
I've only just started taking a look at this repo(we may have some more resources to help out soon). I don't see anything of note from a quick glance but I would like to run locally before approving. I'll try to get to it by mid next week. Thanks again!

File: library/build.gradle [hunk 4]
```diff
@@ -34,6 +35,9 @@ android {
         installOptions "-d", "-t"
     }
 
+    defaultConfig {
+        testInstrumentationRunner 'android.support.test.runner.AndroidJUnitRunner'
```
Comment (friederbluemle) [inline library/build.gradle]: There already is a `defaultConfig` section in this file. Can you merge it with the existing one?

**Comment by friederbluemle** [comment]
@jpoehnelt - Merging the base branch into the PR head branch won't help here - There are a few code changes required to properly fix the test setup on the latest `master` branch. Should I open a PR?

**Comment by ming13** [comment]
Hm, seems like Travis passed, but seems like it doesn’t run tests for now.

**Comment by ming13** [comment]
@friederbluemle, since you’ve resolved test issues separately — I think it would be better to merge your changes first and I’ll rebase mine on top. Also — can I ask to add the `test` Gradle task in Travis configuration on your branch? Don’t want to break anything and it is better to have CI check that.

**Comment by friederbluemle** [comment]
@ming13 - Sure, I'll open a PR with just the test setup fixes. 👍 
I definitely agree this is something that should run on CI with every PR. There are more test updates and fixes on the way, hopefully we'll get there soon. Right now, running the `test` task wouldn't do much more than the existing `check` task does. The tests are setup as "instrumented" tests, and require a connected Android device (something that unfortunately is not that straight forward on Travis CI).

**Comment by ming13** [comment]
@friederbluemle, `testDebugUnitTest` then? I think that tests both of us fixed can be run on the JVM.

**Comment by friederbluemle** [comment]
@ming13 - Yes, you are right, it _should_ be possible to run the tests on the JVM only. There are no real "UI" tests involved. However, since the code tested does contain imports from the Android framework, it may require non trivial additional changes to the test setup, so it might be better to have a separate PR for that.

**Comment by friederbluemle** [comment]
Good news @ming13 - Your branch rebases cleanly onto the latest `master` (with #550 merged). All you need to do is remove the `defaultConfig` block from build.gradle. :)

File: library/build.gradle (hunk 4 above)
Comment (ming13) [reply library/build.gradle]: Removed.

**Comment by ming13** [comment]
@jpoehnelt, @friederbluemle, rebased and squashed! Tests passed locally: 184 of 184.

**Comment by friederbluemle** [comment]
Awesome, thanks @ming13 👏

**Pull request merged**